draft-ietf-jsonpath-base.md

```diff
@@ -1408,15 +1408,15 @@ is a subset of the set of instances of the other type.
 {{tbl-types}} defines the available types in terms of abstract instances, where `n` denotes a node, `v` denotes a value, and `nl` denotes
 a non-empty nodelist. The table also lists the subtypes of each type.
 
-| Type                  | Abstract Instances                       | Subtypes                                 |
-| :--:                  | :----------------:                       | :------:                                 |
-| `OptionalNodeOrValue` | `Node(n)`, `Value(v)`, `Nothing`         | `OptionalNode`, `OptionalValue`, `Value` |
-| `OptionalNode`        | `Node(n)`, `Nothing`                     |                                          |
-| `OptionalValue`       | `Value(v)`, `Nothing`                    | `Value`, `OptionalBoolean`               |
-| `Value`               | `Value(v)`                               | `Boolean`                                |
-| `OptionalBoolean`     | `Value(true)`, `Value(false)`, `Nothing` | `Boolean`                                |
-| `Boolean`             | `Value(true)`, `Value(false)`            |                                          |
-| `OptionalNodes`       | `Nodes(nl)`, `Nothing`                   | `OptionalNode`                           |
+| Type                  | Abstract Instances                       | Subtypes                                                               |
+| :--:                  | :----------------:                       | :------:                                                               |
+| `OptionalNodeOrValue` | `Node(n)`, `Value(v)`, `Nothing`         | `OptionalNode`, `OptionalValue`, `OptionalBoolean`, `Value`, `Boolean` |
```

Comment:
Of course, we can put the transitive closure into the table, but we also could simply say a subtype of a subtype is a subtype.

Reply:
I thought it was clearer to put the transitive closure into the table. I guess another option is to change the column title to "Immediate subtypes" (or similar) to try to avoid confusion. But since the transitive closure isn't massive, I don't think omitting the transitive closure from the table is worth the extra explanation.

We should, however, check that the spec says that a subtype of a subtype is a subtype!

Reply:
To me those additional entries are clutter that make it harder to check and understand the table.
They are probably the reason I didn't see the subtype by coercion thing that I still don't like.

Reply:
Ok, I'll change the table to include only "direct" subtypes and ensure the text says that the subtype relationship is transitive.

Reply:
@cabo Fixed. PTAL.